libs/wingc/src/jsify.rs

```diff
@@ -1057,6 +1070,73 @@ impl<'a> JSifier<'a> {
 		code
 	}
 
+	fn jsify_inflight_init(&self, func_def: &FunctionDefinition, class_phase: Phase, ctx: &mut JSifyContext) -> String {
+		assert!(ctx.visit_ctx.current_phase() == Phase::Inflight);
+
+		let FunctionBody::Statements(body_scope) = &func_def.body else {
+			panic!("inflight init must have a scope body")
+		};
+
+		// Create the async init function that'll capture the ctor's args
+		let mut async_init_body_code = CodeMaker::default();
+		// Define this as a closure if we're inside a regulat ctor (inflight class)
+		async_init_body_code.open(if class_phase == Phase::Inflight {
+			// A closure that'll capture the ctor args
+			format!("this.{CLASS_INFLIGHT_INIT_NAME} = async () => {{")
+		} else {
+			// Preflight class's inflight inits have no args
+			format!("async {CLASS_INFLIGHT_INIT_NAME}() {{")
+		});
+		async_init_body_code.add_code(self.jsify_scope_body(body_scope, ctx));
+		async_init_body_code.close("}");
+
+		// If this is an inflight init of an inflight class then we also need to generate a normat ctor, if it's a preflight class
+		// then we generate a binding ctor seperately see `jsify_inflight_binding_constructor`
+		let code = if class_phase == Phase::Inflight {
+			let mut code = CodeMaker::default();
+			let mut parameter_list = vec![];
+
+			for p in &func_def.signature.parameters {
+				if p.variadic {
+					parameter_list.push("...".to_string() + &p.name.to_string());
+				} else {
+					parameter_list.push(p.name.to_string());
+				}
+			}
+
+			let parameters = parameter_list.iter().map(|x| x.as_str()).collect::<Vec<_>>().join(", ");
```

Comment:
Do we have a `jsify_parameters()` function available?

Reply:
:+1: added